src/slurmd_ops.py

```diff
@@ -0,0 +1,411 @@
+
+    def write_munge_key(self, munge_key: str) -> None:
+        """Base64 decode and write the munge key."""
+        key = b64decode(munge_key.encode())
+        self._munge_key_path.write_bytes(key)
+
+    def version(self) -> str:
+        """Return slurm version."""
+        return Slurmd().version()
+
+    # Fluentbit
+    def fluentbit_config_nhc(self, cluster_name: str, app_name: str) -> list:
+        """Return Fluentbit configuration parameters to forward NHC logs."""
+        cfg = [
+            {
+                "input": [
+                    ("name", "tail"),
+                    ("path", "/var/log/nhc.log"),
+                    ("path_key", "filename"),
+                    ("tag", "nhc"),
+                    ("multiline.parser", "nhc"),
+                ]
+            },
+            {
+                "multiline_parser": [
+                    ("name", "nhc"),
+                    ("type", "regex"),
+                    ("flush_timeout", "1000"),
+                    ("rule", '"start_state"', '"/^([\d]{8} [\d:]*) (.*)/"', '"cont"'),  # noqa
+                    ("rule", '"cont"', r'"/^([^\d].*)/"', '"cont"'),
+                ]
+            },  # noqa
+            {
+                "filter": [
+                    ("name", "record_modifier"),
+                    ("match", "nhc"),
+                    ("record", "hostname ${HOSTNAME}"),
+                    ("record", f"cluster-name {cluster_name}"),
+                    ("record", "service nhc"),
+                    ("record", f"partition-name {app_name}"),
+                ]
+            },
+        ]
+        return cfg
+
+    def fluentbit_config_slurm(self, cluster_name: str, app_name: str) -> list:
+        """Return Fluentbit configuration parameters to forward Slurm logs."""
+        log_file = self._slurmd_log_file
+
+        cfg = [
+            {
+                "input": [
+                    ("name", "tail"),
+                    ("path", log_file.as_posix()),
+                    ("path_key", "filename"),
+                    ("tag", "slurmd"),
+                    ("parser", "slurm"),
+                ]
+            },
+            {
+                "parser": [
+                    ("name", "slurm"),
+                    ("format", "regex"),
+                    ("regex", r"^\[(?<time>[^\]]*)\] (?<log>.*)$"),
+                    ("time_key", "time"),
+                    ("time_format", "%Y-%m-%dT%H:%M:%S.%L"),
+                ]
+            },
+            {
+                "filter": [
+                    ("name", "record_modifier"),
+                    ("match", "slurmd"),
+                    ("record", "hostname ${HOSTNAME}"),
+                    ("record", f"cluster-name {cluster_name}"),
+                    ("record", f"partition-name {app_name}"),
+                    ("record", "service slurmd"),
+                ]
+            },
+        ]
+        return cfg
+
+    # NHC
+    def _install_nhc_from_tarball(self) -> bool:
```

Comment:
I thought about this ... we could if we wanted to install it in the charm_dir. I'm game for it!

Reply:
Great! We should look into it for a follow-on pull request. One thing I'd be interested to know is whether we can emulate the desired behavior of NHC with Prometheus + Alert Manager. I'm interested to know what the overlap is versus what gaps there are.

Also, we could look into packaging NHC so that we don't even have to worry about building it ourselves.